sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -239,10 +247,10 @@ def test_affinity_propagation_random_state():
 
 
 @pytest.mark.parametrize("centers", [csr_matrix(np.zeros((1, 10))), np.zeros((1, 10))])
-def test_affinity_propagation_convergence_warning_dense_sparse(centers):
+def test_affinity_propagation_convergence_warning_dense_sparse(centers, global_dtype):
     """Non-regression, see #13334"""
     rng = np.random.RandomState(42)
-    X = rng.rand(40, 10)
+    X = rng.rand(40, 10).astype(global_dtype, copy=False)
     y = (4 * rng.rand(40)).astype(int)
     ap = AffinityPropagation(random_state=46)
     ap.fit(X, y)
```

Comment:
I assume that the test below that is a regression for 32 bits could change and run on both 64 and 32 bits?

Reply:
I mean the `test_affinity_propagation_float32`.